python/pyarrow/parquet.py

```diff
@@ -845,16 +862,18 @@ class ParquetDataset(object):
     def __init__(self, path_or_paths, filesystem=None, schema=None,
                  metadata=None, split_row_groups=False, validate_schema=True,
                  filters=None, metadata_nthreads=1):
-        if filesystem is None:
-            a_path = path_or_paths
-            if isinstance(a_path, list):
-                a_path = a_path[0]
-            self.fs = _get_fs_from_path(a_path)
-        else:
-            self.fs = _ensure_filesystem(filesystem)
+        a_path = path_or_paths
+        if isinstance(a_path, list):
+            a_path = a_path[0]
 
+        self.fs, _ = _get_filesystem_and_path(filesystem, a_path)
         self.paths = path_or_paths
```

Comment:
This temporary assignment looks a bit superfluous. You could just access `path_or_paths` below.

Reply:
fixed